setup.py

```diff
@@ -39,7 +39,7 @@ def read_file(path_segments):
         "canonicaljson>=1.0.0",
         "unpaddedbase64>=1.0.1",
         "pynacl>=0.3.0",
-        "typing_extensions>=3.5",
+        'typing_extensions>=3.5;python_version<"3.8"',
         'typing>=3.5;python_version<"3.5"',
         'importlib_metadata;python_version<"3.8"',
     ],
```

Comment:
I think we always need `typing_extensions` as a dev dependency now.

```suggestion
    ],
    extras_require={
        "dev": ["typing_extensions>=3.5"],
    },
```

Reply:
Is that because the linters would find an unmet dependency?

Reply:
Yes, when running mypy, we'd always use `typing_extensions` now